### DISCUSSION TITLE
multiple endpoints for OIDCRedirectURI

### DISCUSSION THREAD

**Discussion by sivakura** [OpenIDC/mod_auth_openidc#1041] [Q&A] [answered]
hi there,

My environment has one apache serving multiple apps. To use okta, now configuring mod_auth_openidc.
Below is my setup with authenticated apps and non authenticated apps.

https://test.com/abc - need authentication
https://test.com/def - need authentication
https://test.com/xyz - no authentication

how can i configure multiple redirecturi in for "OIDCRedirectURI" in auth_openidc.conf 

Thanks,
Siva

**Comment by zandbelt** [comment, marked as answer]
you don't need to define multiple redirect URIs; your use case is catered for with the regular setup with one redirect URI; locations that need authentication should be protected with `AuthType openid-connect`

**Reply by sivakura** [reply]
Thank you!!

**Comment by sealor** [comment]
Hello @zandbelt ,

I try to setup the following locations:

- https://test.com/abc <- secured by OpenIDC
- https://test.com/def <- secured by OpenIDC
- https://test.com/xyz <- no auth
- https://test.com/index.html <- no auth

How can I get this to work?

Currently, I get redirected to <https://test.com/?code=...&iss=...> when trying to open one of the secured apps. I'm using `OIDCRedirectURI /`.
Both secured apps work when adding `AuthType openid-connect` to `<Location />` but this is not what we need because then https://test.com/index.html is secured.

Best regards,
Stefan

**Reply by zandbelt** [reply]
just set `OIDCRedirectURI` to `/abc/redirect_uri`
